### DISCUSSION TITLE
Installation error

### DISCUSSION THREAD

**Discussion by AbdelAziz-Maysara** [projectdiscovery/nuclei#856] [Q&A] [answered]
**Describe the bug**
I have used the script GO111MODULE=on go get -v github.com/projectdiscovery/nuclei/v2/cmd/nuclei and downloading begins then at last it shows reference errors 
/usr/bin/ld: /home/prince/.cache/go-build/de/de3155f24a950d9ea059b65bfe4749561a665dd0935d3c1dfec534d4d8dfe188-d(_go_.o): in function `github.x2ecom..z2fmodern..z2dgo..z2freflect2.loadGo15Types':
/home/prince/go/pkg/mod/github.com/modern-go/reflect2@v1.0.1/type_map.go:42: undefined reference to `reflect.typelinks'
/usr/bin/ld: /home/prince/.cache/go-build/de/de3155f24a950d9ea059b65bfe4749561a665dd0935d3c1dfec534d4d8dfe188-d(_go_.o): in function `github.x2ecom..z2fmodern..z2dgo..z2freflect2.discoverTypes':
/home/prince/go/pkg/mod/github.com/modern-go/reflect2@v1.0.1/type_map.go:74: undefined reference to `reflect.typelinks'
/usr/bin/ld: /home/prince/.cache/go-build/de/de3155f24a950d9ea059b65bfe4749561a665dd0935d3c1dfec534d4d8dfe188-d(_go_.o): in function `reflect2.loadGo17Types':
/home/prince/go/pkg/mod/github.com/modern-go/reflect2@v1.0.1/type_map.go:78: undefined reference to `reflect.resolveTypeOff'
collect2: error: ld returned 1 exit status

**Nuclei version**
Please share the version of the nuclei you are running with `nuclei -version` V 2.4.0

**Screenshot of the error or bug**
please add the screenshot showing bug or issue you are facing.

**Comment by geeknik** [comment]
Works fine here on both Fedora 34 and Ubuntu 20.04... It could be something related to your Go setup? What version of Go do you have installed? We recommend 1.16 or higher, as 1.16.6 is the latest release.

**Comment by AbdelAziz-Maysara** [comment]
I have a kali virtual machine and the version installed is 1.14.6 and I can't upgrade it

**Comment by Mzack9999** [comment, marked as answer]
According to the output, you should already have the latest version (`2.4.0`). You could try to clean up the go cache with:

```
go clean --modcache
```

As an alternative, another option is to download the compiled binary at https://github.com/projectdiscovery/nuclei/releases
